droidlet/lowlevel/minecraft/craftassist_mover.py

```diff
@@ -0,0 +1,254 @@
+        self.get_other_players = struct_transform(self.cagent.get_other_players)
+        self.get_other_player_by_name = struct_transform(self.cagent.get_other_player_by_name)
+
+    @struct_transform
+    def get_player_line_of_sight(self, player_struct):
+        # this is a little tricky: the player_struct in droidlet space has pos x-flipped
+        # and look yaw and pitch flipped.  we need the cagent's player struct to do the computation in
+        # cuberite.  If the player_struct is a python object from droidlet, we assume it has a
+        # cuberite cagent player struct as a member.
+        if hasattr(player_struct, "cagent_struct"):
+            return self.cagent.get_player_line_of_sight(player_struct.cagent_struct)
+        else:
+            # TODO expose cagent's player struct def and assert that data type is correct
+            return self.cagent.get_player_line_of_sight(player_struct)
+
+    def dig(self, x, y, z):
+        self.cagent.place_block(-x, y, z)
+
+    def place_block(self, x, y, z):
+        self.cagent.place_block(-x, y, z)
+
+    def get_changed_blocks(self):
+        blocks = self.cagent.get_changed_blocks()
+        transformed_blocks = []
+        for xyz, idm in blocks:
+            transformed_blocks.append(((-xyz[0], xyz[1], xyz[2]), idm))
+        return transformed_blocks
+
+    # FIXME!! turn_angle is broken in the cagent; should be swapping here,
+    # but cagent actually has it backwards
+    # turn_angle isn't being used in other parts of agent.
+    def turn_angle(self, yaw):
+        self.cagent.turn_angle(yaw)
+
+    def set_look(self, yaw, pitch):
+        self.cagent.set_look(-yaw, pitch)
+
+    def look_at(self, x, y, z):
+        self.cagent.look_at(-x, y, z)
+
+    def get_blocks(self, x, X, y, Y, z, Z):
+        """
+        returns an (X-x+1) x (Y-y+1) x (Z-z+1) x 2 numpy array B of the blocks
+        in the rectanguloid with bounded by the input coordinates (including endpoints).
+        Input coordinates are in droidlet coordinates; and the output array is
+        in yzxb permutation, where B[0,0,0,:] corresponds to the id and meta of
+        the block at x, y, z
+        """
+        # negate the x coordinate to shift to cuberite coords
+        B = self.cagent.get_blocks(-X, -x, y, Y, z, Z)
+        return np.flip(B, 2)
+
+    # reversed to match droidlet coords
+    def step_pos_x(self):
+        self.cagent.step_neg_x()
+
+    # reversed to match droidlet coords
+    def step_neg_x(self):
+        self.cagent.step_pos_x()
+
+    def step_pos_y(self):
+        self.cagent.step_pos_y()
+
+    def step_neg_y(self):
+        self.cagent.step_neg_y()
+
+    def step_pos_z(self):
+        self.cagent.step_pos_z()
+
+    def step_neg_z(self):
+        self.cagent.step_neg_z()
+
+    def step_forward(self):
+        self.cagent.step_forward()
+
+
+############################################################################
+# in minecraft, we have
```

Comment:
Ah just saw that the documentation is in this file, thanks for adding a mover. Might still be worth linking this file from main README for ca agent